src/main/scala/com/microsoft/hyperspace/index/rules/RuleUtils.scala

```diff
@@ -259,10 +262,18 @@ object RuleUtils {
 
         val (filesDeleted, filesAppended) =
           if (HyperspaceConf.hybridScanDeleteEnabled(spark) && index.hasLineageColumn(spark)) {
-            val (exist, nonExist) = curFileSet.partition(index.sourceFileInfoSet.contains)
+            val sourceFileInfos = // remove file id so comparison only uses actual file properties.
```

Comment:
Question: isn't it possible to define(override) equals function(comparison function?) not considering id field?

Reply:
That is a good point, I had thought about it. The issue is that there are cases where we need to consider file id when comparing two fileInfos and cases we dont. Basically, we need two equal functions.

Reply:
Do we need to distinguish the FileInfos with the same path/size/modification time? I think we always compare FileInfo with Set[FileInfo] which "name" is fullPath in most cases?

Reply:
Currently, with this change, in `RefreshActionBase` when computing deletedFiles, id is also part of comparison. However, I got your point. Physical file properties should be enough. I will add `equals` to FileInfo and ignore id for the comparison and verify if everything works fine or not.